internal/pkg/api/handleCheckin.go

```diff
@@ -90,6 +92,13 @@ func NewCheckinT(
 	tr *action.TokenResolver,
 	bulker bulk.Bulk,
 ) *CheckinT {
+	rt := rate.Every(cfg.Limits.ActionLimit.Interval)
+	if cfg.Limits.ActionLimit.Interval == 0 && cfg.Limits.PolicyThrottle != 0 {
+		rt = rate.Every(cfg.Limits.PolicyThrottle)
+	} else if cfg.Limits.ActionLimit.Interval == 0 && cfg.Limits.PolicyThrottle == 0 {
+		rt = rate.Inf
+	}
+	zerolog.Ctx(context.TODO()).Debug().Any("event_rate", rt).Int("burst", cfg.Limits.ActionLimit.Burst).Msg("checkin response gzip limiter")
```

Comment:
Ideally yes, but we would need to make sure that the context is tied to the function instead of `checkinT`'s lifecycle.
We also use context.TODO in a few other places similar to this.

Reply:
There's an existing issue to address this: https://github.com/elastic/fleet-server/issues/3087